### PR TITLE
SAIC-97 To avoid DHCP taking IP used by an instance

### PR DESCRIPTION
**Pull request by wzbbbb**
Since the creation order of a DHCP port and an instance port can be
random, sometimes, the DHCP port can take the IP used by an instance
in source could. To avoid it, this fix will disable DHCP and free its
IP before instance port creation, then re-enable the DHCP.
	modified:   cloudferrylib/os/actions/prepare_networks.py
	modified:   cloudferrylib/os/network/neutron.py

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/168/

File: cloudferrylib/os/actions/prepare_networks.py [hunk 1]
```diff
@@ -27,10 +27,15 @@ def run(self, info=None, **kwargs):
 
         network_resource = self.cloud.resources[utl.NETWORK_RESOURCE]
         identity_resource = self.cloud.resources[utl.IDENTITY_RESOURCE]
-
         keep_ip = self.cfg.migrate.keep_ip
-
         instances = info_compute[utl.INSTANCES_TYPE]
+        #disable DHCP in all subnets
+        subnets=network_resource.get_subnets()
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/prepare_networks.py]: pep8 violation:
C: 33, 0: Exactly one space required around assignment
        subnets=network_resource.get_subnets()
               ^ (bad-whitespace)

File: cloudferrylib/os/actions/prepare_networks.py [hunk 2]
```diff
@@ -59,12 +64,22 @@ def run(self, info=None, **kwargs):
...
                         dst_flotingips = network_resource.get_floatingips()
                         dst_flotingips_map = \
-                            {fl_ip['floating_ip_address']: fl_ip['id'] for fl_ip in dst_flotingips}
-                        dst_floatingip_id = dst_flotingips_map[src_net['floatingip']]
-                        floating_ip = network_resource.update_floatingip(dst_floatingip_id, port['id'])
+                            {fl_ip['floating_ip_address']: \
+                                fl_ip['id'] for fl_ip in dst_flotingips}
+                        dst_floatingip_id = \
+                            dst_flotingips_map[src_net['floatingip']]
+                        floating_ip = \
```
Comment (roman-verchikov) [inline cloudferrylib/os/actions/prepare_networks.py]: Unused variable `floating_ip`

**Comment by roman-verchikov** [comment]
Just a general recommendation: please run `pylint`, `pep8` and `flake8` *before* pushing changes upstream

File: cloudferrylib/os/actions/prepare_networks.py (hunk 2 above)
Comment (wzbbbb) [inline cloudferrylib/os/actions/prepare_networks.py]: That's from the upstream. I did not modify that part. All I did was breaking them to shorter lines.

Comment (roman-verchikov) [inline cloudferrylib/os/actions/prepare_networks.py]: You shouldn't touch old code unless you introduce new or modify existing functionality. This way it's easier both for reviewers (less code to look through) and for you (less code to modify). In this particular case: 
1. Undo all the non-functional changes you made (whitespace, indentation, line breaks, etc)
2. Leave only functional changes and verify they don't break pep8

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/175/

**Comment by wzbbbb** [comment]
Cleaned up a bit the 2 files. Did not modify the unused variables part.

**Pull request closed without merging**